src/nbl/asset/utils/IShaderCompiler.cpp

```diff
@@ -224,7 +294,7 @@ auto IShaderCompiler::CIncludeFinder::tryIncludeGenerators(const std::string& in
 
 core::smart_refctd_ptr<asset::ICPUShader> IShaderCompiler::CCache::find(const SEntry& mainFile, const IShaderCompiler::CIncludeFinder* finder) const
 {
-    return find_impl(mainFile, finder)->cpuShader;
+    return find_impl(mainFile, finder)->decodeShader();
```

Comment:
ooof, seems like nobody checked (even before) if the thing was even found before dereferencing an iterator

Reply:
this needs fixing, do 
```cpp
const auto found = find_impl(mainFile,finder);
if (found==m_container.end())
   return nullptr;
return found->decodeShader();
```

Reply:
yeah i forgot this review